lib/bin/lwm2m_carrier/os/lwm2m_os.c

```diff
@@ -367,7 +397,7 @@ int lwm2m_os_thread_start(int index, lwm2m_os_thread_entry_t entry, const char *
 int lwm2m_os_nrf_modem_init(void)
 {
 #if defined CONFIG_NRF_MODEM_LIB_SYS_INIT
-	int nrf_err = modem_lib_init_result;
+	int nrf_err = nrf_modem_lib_get_init_ret();
```

Comment:
This is deprecated, you should keep using `NRF_MODEM_LIB_ON_INIT` to define a callback instead.

Reply:
Thanks